nodespec/azure.go

```diff
@@ -0,0 +1,263 @@
+  name: Standard_E8s_v3
+  numberOfCores: 8
+  osDiskSizeInMb: 1047552
+  resourceDiskSizeInMb: 137438.953472
+Standard_E16s_v3:
+  descriptions: Esv3-series, memory optimized, 160-190 ACU, premium storage supported
+  additionalProperties: {}
+  maxDataDiskCount: 32
+  memoryInMb: 137438.953472
+  name: Standard_E16s_v3
+  numberOfCores: 16
+  osDiskSizeInMb: 1047552
+  resourceDiskSizeInMb: 274877.906944
+Standard_E32s_v3:
+  description: Esv3-series, memory optimized, 160-190 ACU, premium storage supported
+  additionalProperties: {}
+  maxDataDiskCount: 32
+  memoryInMb: 274877.906944
+  name: Standard_E32s_v3
+  numberOfCores: 32
+  osDiskSizeInMb: 1047552
+  resourceDiskSizeInMb: 549755.813888
+Standard_F4s_v2:
+  description: Fsv2-series, compute optimized, 195-210 ACU, premium storage supported
+  additionalProperties: {}
+  maxDataDiskCount: 8
+  memoryInMb: 8589.934592
+  name: Standard_F4s_v2
+  numberOfCores: 4
+  osDiskSizeInMb: 1047552
+  resourceDiskSizeInMb: 34359.738368
+Standard_F8s_v2:
+  description: Fsv2-series, compute optimized, 195-210 ACU, premium storage supported
+  additionalProperties: {}
+  maxDataDiskCount: 16
+  memoryInMb: 17179.869184
+  name: Standard_F8s_v2
+  numberOfCores: 8
+  osDiskSizeInMb: 1047552
+  resourceDiskSizeInMb: 68719.476736
+Standard_F16s_v2:
+  description: Fsv2-series, compute optimized, 195-210 ACU, premium storage supported
+  additionalProperties: {}
+  maxDataDiskCount: 32
+  memoryInMb: 34359.738368
+  name: Standard_F16s_v2
+  numberOfCores: 16
+  osDiskSizeInMb: 1047552
+  resourceDiskSizeInMb: 137438.953472
+Standard_F32s_v2:
+  description: Fsv2-series, compute optimized, 195-210 ACU, premium storage supported
+  additionalProperties: {}
+  maxDataDiskCount: 32
+  memoryInMb: 68719.476736
+  name: Standard_F32s_v2
+  numberOfCores: 32
+  osDiskSizeInMb: 1047552
+  resourceDiskSizeInMb: 274877.906944
+`
+)
+
+// ProviderAzure contains all provider specific info
+type ProviderAzure struct {
+	vmSizes map[string]VMSize
+}
+
+type VMSize struct {
+	Description          string  `yaml:"description"`
+	MaxDataDiskCount     int     `yaml:"maxDataDiskCount"`
+	MemoryInMB           float64 `yaml:"memoryInMb"`
+	Name                 string  `yaml:"name"`
+	NumberOfCores        int64   `yaml:"numberOfCores"`
+	OSDiskSizeInMB       int64   `yaml:"osDiskSizeInMb"`
+	ResourceDiskSizeInMB float64 `yaml:"resourceDiskSizeInMb"`
+}
+
+// NewAzureProvider initiates a new Azure provider with the information about VM sizes.
+func NewAzureProvider() (*ProviderAzure, error) {
+	p := &ProviderAzure{}
+
+	err := yaml.Unmarshal([]byte(azureVMSizesYAML), &p.vmSizes)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	return p, nil
+}
+
+// GetVMSizeDetails returns info on a certain VM size
+func (p *ProviderAzure) GetVMSizeDetails(name string) (*VMSize, error) {
+	vmSize, ok := p.vmSizes[name]
+	if ok {
+		return &vmSize, nil
+	}
+
+	fmt.Printf("VM size not found %q", name)
```

Comment:
We usually keep error printing into a single place, as it's much easier to deal with if you want to change something. You can handle the error printing here, instead
https://github.com/giantswarm/gsctl/blob/30bf71c831ca4640523cf7a7180a257ebf22ccd4/commands/list/nodepools/command.go#L386

Reply:
Damn, this was a debugging leftover, thanks for pointing out :)